aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -117,14 +117,19 @@ void startSubscriptions() {
                 clazz,
                 token -> latch.countDown(),
                 emitter::onNext,
-                emitter::onError,
+                AmplifyDisposables.onErrorConsumerWrapperFor(emitter),
                 emitter::onComplete
             );
             // When the observable is disposed, we need to call cancel() on the subscription
             // so it can properly dispose of resources if necessary. For the AWS API plugin,
             // this means means closing the underlying network connection.
             emitter.setDisposable(AmplifyDisposables.fromCancelable(cancelable));
-            latch.await(SUBSCRIPTION_START_TIMEOUT_MS, TimeUnit.MILLISECONDS);
```

Comment:
Even before, we weren't checking the return value here -- wether or not the latch actually counted down. Should we be, now?